Jenkinsfile

```diff
@@ -43,458 +60,606 @@ pipeline {
           }
           agent { label 'armv7' }
           steps {
             script {
-              debugBuild = load ".jenkinsci/debug-build.groovy"
-              coverage = load ".jenkinsci/selected-branches-coverage.groovy"
-              if (!params.x86_64_linux && !params.armv8_linux && !params.x86_64_macos && (coverage.selectedBranchesCoverage(['develop', 'master']))) {
-                debugBuild.doDebugBuild(true)
+              if ( params.BUILD_TYPE == 'Debug' || params.Merge_PR ) {
+                def debugBuild = load ".jenkinsci/debug-build.groovy"
+                def coverage = load ".jenkinsci/selected-branches-coverage.groovy"
+                debugBuild.doDebugBuild( (!params.Linux && !params.MacOS && !params.ARMv8 || !params.Merge_PR) ? coverage.selectedBranchesCoverage() : false )
               }
               else {
-                debugBuild.doDebugBuild()
-              }
-              if (GIT_LOCAL_BRANCH ==~ /(master|develop)/) {
-                releaseBuild = load ".jenkinsci/release-build.groovy"
+                def releaseBuild = load ".jenkinsci/release-build.groovy"
                 releaseBuild.doReleaseBuild()
               }
             }
           }
           post {
-            always {
+            success {
               script {
-                post = load ".jenkinsci/linux-post-step.groovy"
-                post.linuxPostStep()
+                if (params.BUILD_TYPE == 'Release') {
+                  def post = load ".jenkinsci/linux-post-step.groovy"
+                  post.postStep()
+                }
               }
             }
           }
         }
-        stage('armv8_linux') {
+        stage('ARMv8') {
           when {
             beforeAgent true
-            expression { return params.armv8_linux }
+            anyOf {
+              expression { return params.ARMv8 }
+              // allOf {
+              //   expression { return env.CHANGE_ID != null }
+              //   expression { return env.GIT_PREVIOUS_COMMIT != null }
+              //   expression { return env.CHANGE_TARGET ==~ /(master|develop|trunk)/ }
+              //   expression { return params.Merge_PR }
+              // }
+            }
           }
           agent { label 'armv8' }
           steps {
             script {
-              debugBuild = load ".jenkinsci/debug-build.groovy"
-              coverage = load ".jenkinsci/selected-branches-coverage.groovy"
-              if (!params.x86_64_linux && !params.x86_64_macos && (coverage.selectedBranchesCoverage(['develop', 'master']))) {
-                debugBuild.doDebugBuild(true)
+              if ( params.BUILD_TYPE == 'Debug' || params.Merge_PR ) {
+                def debugBuild = load ".jenkinsci/debug-build.groovy"
+                def coverage = load ".jenkinsci/selected-branches-coverage.groovy"
+                debugBuild.doDebugBuild( (!params.Linux && !params.MacOS || !params.Merge_PR) ? coverage.selectedBranchesCoverage() : false )
```

Comment:
This conditional could be removed because of reasons above, same for calls for `mac`, `armv7`.

Reply:
Thanks, now we can get rid of these complex conditions